### PR TITLE
Start with a Guided Tour.

### PR DESCRIPTION
**Pull request by mohsenD98**
### Enhanced First-Time User Experience (FTUE)

This pull request introduces a guided tutorial for new users when they launch the application for the first time. The goal is to provide an intuitive onboarding experience, setting the stage for a successful and enjoyable usage of Qfield.

**Demonstration**:
To illustrate the new feature, please watch the demo:    
   
**Dark**   
   

[Screencast from 2024-07-25 11-46-19.webm](https://github.com/user-attachments/assets/8cdac7fc-cd04-4ac4-9aa6-dfbbb8217322)    

    

**Light**

   
[Screencast from 2024-07-25 11-46-52.webm](https://github.com/user-attachments/assets/e4adf855-a469-4b73-bc14-c1af627f6605)

As shown in the demo, we have implemented a tutorial panel that introduces key features step-by-step. The guide is designed to gradually introduce users to the application's functionality.

In this initial implementation, we've focused on providing essential information to new users. Future enhancements will further refine the tutorial to offer an even more engaging and effective learning experience.

This update aims to simplify the learning curve for our users, ensuring they can quickly get started with Qfield and maximize their productivity.

### CODE REVIEW

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 51db6923ed72cf541772207733a555068ff850f0: [**arm64-android**](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield-dev-5465-eabf3b2ec725de185f66aea5901b400691553b38-android-arm64.apk)
<details>
<summary>Other architectures</summary>

- [arm-neon-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield-dev-5465-eabf3b2ec725de185f66aea5901b400691553b38-android-arm-neon.apk)
- [x64-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield-dev-5465-eabf3b2ec725de185f66aea5901b400691553b38-android-x64.apk)
- [x86-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield-dev-5465-eabf3b2ec725de185f66aea5901b400691553b38-android-x86.apk)
- [**all access arm64-android**](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield_all_access-dev-5465-eabf3b2ec725de185f66aea5901b400691553b38-android-arm64.apk)
- [all access arm-neon-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield_all_access-dev-5465-eabf3b2ec725de185f66aea5901b400691553b38-android-arm-neon.apk)
- [all access x64-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield_all_access-dev-5465-eabf3b2ec725de185f66aea5901b400691553b38-android-x64.apk)
- [all access x86-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield_all_access-dev-5465-eabf3b2ec725de185f66aea5901b400691553b38-android-x86.apk)
</details>

File: src/qml/WelcomeScreen.qml [hunk 1]
```diff
@@ -884,6 +884,7 @@ Page {
         if (firstRun) {
           welcomeText.text = qsTr("Welcome to QField. First time using this application? Try the sample projects listed below.");
           settings.setValue("/QField/FirstRunDone", true);
+          settings.setValue("/QField/showMapButtonsGuide", true);
```
Comment (nirvn) [inline src/qml/WelcomeScreen.qml:887]:
```suggestion
          settings.setValue("/QField/showMapCanvasGuide", true);
```

File: src/qml/qgismobileapp.qml [hunk 2]
```diff
@@ -3931,4 +3932,38 @@ ApplicationWindow {
...
+    }
+    settings.setValue("/QField/showMapButtonsGuide", false);
+  }
+
+  function runStartupTour() {
+    startupTour.open();
+  }
+
+  QFieldGuide {
+    id: startupTour
```
Comment (nirvn) [inline src/qml/qgismobileapp.qml:3949]:
```suggestion
    id: mapCanvasTour
```

File: src/qml/qgismobileapp.qml [hunk 3]
```diff
@@ -3931,4 +3932,38 @@ ApplicationWindow {
...
+    const startupGuide = settings.valueBool("/QField/showMapButtonsGuide", true);
+    if (startupGuide) {
+      runStartupTour();
+    }
+    settings.setValue("/QField/showMapButtonsGuide", false);
+  }
+
+  function runStartupTour() {
+    startupTour.open();
+  }
```
Comment (nirvn) [inline src/qml/qgismobileapp.qml:3946]: Could we move this within the QFieldGuide {} item below?

File: src/qml/qgismobileapp.qml [hunk 4]
```diff
@@ -3931,4 +3932,38 @@ ApplicationWindow {
...
+
+  QFieldGuide {
+    id: startupTour
+    baseRoot: mainWindow
+    steps: [{
+        "title": qsTr("Zoom Toolbar"),
+        "description": qsTr("Here you can zoom in/out easily through these buttons."),
```
Comment (nirvn) [inline src/qml/qgismobileapp.qml:3953]:
```suggestion
        "description": qsTr("In addition to pinch gesture, these buttons helps you quickly zooming in and out."),
```

File: src/qml/qgismobileapp.qml [hunk 5]
```diff
@@ -3931,4 +3932,38 @@ ApplicationWindow {
...
+        "target": () => [zoomToolbar]
+      }, {
+        "title": qsTr("Menu"),
+        "description": qsTr("You can open dashboard here, see map legends or start digitizing."),
```
Comment (nirvn) [inline src/qml/qgismobileapp.qml:3957]: I think I'd start with this one, that's essentially the primary button.

```suggestion
        "description": qsTr("You can open dashboard here to interact with the project's legends and map theme, or start digitizing by activating the editing mode. Long-pressing the button also gives you immediate access to the main menu."),
```

File: src/qml/qgismobileapp.qml [hunk 6]
```diff
@@ -3931,4 +3932,38 @@ ApplicationWindow {
...
+        "target": () => [menuButton]
+      }, {
+        "title": qsTr("GNSS"),
+        "description": qsTr("Global navigation satellite system provides positioning and navigation."),
```
Comment (nirvn) [inline src/qml/qgismobileapp.qml:3961]:
```suggestion
        "description": qsTr("This button toggles your device's positioning system. When enabled, a position marker will appear top of the map. Long-pressing the button will open the positioning menu where additional functionalities can be explored."),
```

File: src/qml/qgismobileapp.qml [hunk 7]
```diff
@@ -3931,4 +3932,38 @@ ApplicationWindow {
...
+        "target": () => [gnssButton]
+      }, {
+        "title": qsTr("Search"),
+        "description": qsTr("Search and locate on the map."),
```
Comment (nirvn) [inline src/qml/qgismobileapp.qml:3965]:
```suggestion
        "description": qsTr("The search bar provides you with a quick way to find features within your project, jump to a typed latitude and longitude point, and much more."),
```

**Comment by nirvn** [comment]
We need to find a way for these guides not to show up in our testing environment :-)

File: src/qml/WelcomeScreen.qml (hunk 1 above)
Comment (mohsenD98) [reply src/qml/WelcomeScreen.qml]: done :+1:

File: src/qml/qgismobileapp.qml (hunk 2 above)
Comment (mohsenD98) [reply src/qml/qgismobileapp.qml]: done :+1:

File: src/qml/qgismobileapp.qml (hunk 4 above)
Comment (mohsenD98) [reply src/qml/qgismobileapp.qml]: done 👍

File: src/qml/qgismobileapp.qml (hunk 5 above)
Comment (mohsenD98) [reply src/qml/qgismobileapp.qml]: done 👍

File: src/qml/qgismobileapp.qml (hunk 6 above)
Comment (mohsenD98) [reply src/qml/qgismobileapp.qml]: done 👍

File: src/qml/qgismobileapp.qml (hunk 7 above)
Comment (mohsenD98) [reply src/qml/qgismobileapp.qml]: done 👍

**Comment by mohsenD98** [comment]
also i pushed modified UI as recommended in our chat.

**Comment by suricactus** [comment]
Just want to say, fantastic feature! One very small suggestion, I think it would be nice to have the current step we are at  with the dialogs, e.g. GNSS is 3/5. It's always good to know your road ahead.

File: src/qml/qgismobileapp.qml [hunk 8]
```diff
@@ -3931,4 +3932,44 @@ ApplicationWindow {
...
+        "title": qsTr("Zoom Toolbar"),
+        "description": qsTr("In addition to pinch gesture, these buttons helps you quickly zooming in and out."),
+        "target": () => [zoomToolbar]
+      }, {
+        "title": qsTr("GNSS"),
+        "description": qsTr("This button toggles your device's positioning system. When enabled, a position marker will appear top of the map. Long-pressing the button will open the positioning menu where additional functionalities can be explored."),
+        "target": () => [gnssButton]
+      }, {
+        "title": qsTr("Search"),
+        "description": qsTr("The search bar provides you with a quick way to find features within your project, jump to a typed latitude and longitude point, and much more."),
```
Comment (m-kuhn) [inline src/qml/qgismobileapp.qml:3955]: Should we link "and much more" to the docs?

File: src/qml/qgismobileapp.qml [hunk 9]
```diff
@@ -3931,4 +3932,44 @@ ApplicationWindow {
...
+        "title": qsTr("Menu"),
+        "description": qsTr("You can open dashboard here to interact with the project's legends and map theme, or start digitizing by activating the editing mode. Long-pressing the button also gives you immediate access to the main menu."),
+        "target": () => [menuButton]
+      }, {
+        "title": qsTr("Zoom Toolbar"),
+        "description": qsTr("In addition to pinch gesture, these buttons helps you quickly zooming in and out."),
+        "target": () => [zoomToolbar]
+      }, {
+        "title": qsTr("GNSS"),
+        "description": qsTr("This button toggles your device's positioning system. When enabled, a position marker will appear top of the map. Long-pressing the button will open the positioning menu where additional functionalities can be explored."),
```
Comment (m-kuhn) [inline src/qml/qgismobileapp.qml:3951]:
```suggestion
        "description": qsTr("This button toggles the positioning system. When enabled, a position marker will appear top of the map. Long-pressing the button will open the positioning menu where additional functionalities can be explored."),
```

File: src/qml/qgismobileapp.qml [hunk 10]
```diff
@@ -3931,4 +3932,44 @@ ApplicationWindow {
...
+    baseRoot: mainWindow
+    objectName: 'mapCanvasTour'
+
+    steps: [{
+        "title": qsTr("Menu"),
+        "description": qsTr("You can open dashboard here to interact with the project's legends and map theme, or start digitizing by activating the editing mode. Long-pressing the button also gives you immediate access to the main menu."),
+        "target": () => [menuButton]
+      }, {
+        "title": qsTr("Zoom Toolbar"),
+        "description": qsTr("In addition to pinch gesture, these buttons helps you quickly zooming in and out."),
```
Comment (m-kuhn) [inline src/qml/qgismobileapp.qml:3947]:
```suggestion
        "description": qsTr("In addition to the pinch gesture, these buttons help you quickly zoom in and out."),
```

File: src/qml/qgismobileapp.qml [hunk 11]
```diff
@@ -3931,4 +3932,44 @@ ApplicationWindow {
...
   }
+
+  QFieldGuide {
+    id: mapCanvasTour
+    baseRoot: mainWindow
+    objectName: 'mapCanvasTour'
+
+    steps: [{
+        "title": qsTr("Menu"),
+        "description": qsTr("You can open dashboard here to interact with the project's legends and map theme, or start digitizing by activating the editing mode. Long-pressing the button also gives you immediate access to the main menu."),
```
Comment (m-kuhn) [inline src/qml/qgismobileapp.qml:3943]:
```suggestion
        "description": qsTr("This button opens the dashboard. With the dashboard you can interact with the legend and map theme, or start digitizing by activating the editing mode. Long-pressing the button gives you immediate access to the main menu."),
```

**Comment by m-kuhn** [comment]
Looking good !
I just left a couple of small wording improvements.

Is it possible to abort the tour? Someone who sets up 50 tablets for a team should not have to click through.

**Comment by mohsenD98** [comment]
> Looking good ! I just left a couple of small wording improvements.
> 
> Is it possible to abort the tour? Someone who sets up 50 tablets for a team should not have to click through.

Thankyou, yes its possible:  
  

image

**Comment by mohsenD98** [comment]
> Just want to say, fantastic feature! One very small suggestion, I think it would be nice to have the current step we are at with the dialogs, e.g. GNSS is 3/5. It's always good to know your road ahead.

Really nice idea ! working on it :)  :+1:

File: src/qml/qgismobileapp.qml (hunk 11 above)
Comment (mohsenD98) [reply src/qml/qgismobileapp.qml:3943]: :+1:

File: src/qml/qgismobileapp.qml [hunk 12]
```diff
@@ -3931,4 +3932,44 @@ ApplicationWindow {
...
     }
   }
+
+  QFieldGuide {
+    id: mapCanvasTour
+    baseRoot: mainWindow
+    objectName: 'mapCanvasTour'
+
+    steps: [{
+        "title": qsTr("Menu"),
```
Comment (nirvn) [inline src/qml/qgismobileapp.qml:3942]:
```suggestion
        "title": qsTr("Dashboard"),
```

File: src/qml/qgismobileapp.qml [hunk 13]
```diff
@@ -3931,4 +3932,44 @@ ApplicationWindow {
...
+        "description": qsTr("This button opens the dashboard. With the dashboard you can interact with the legend and map theme, or start digitizing by activating the editing mode. Long-pressing the button gives you immediate access to the main menu."),
+        "target": () => [menuButton]
+      }, {
+        "title": qsTr("Zoom Toolbar"),
```
Comment (nirvn) [inline src/qml/qgismobileapp.qml:3946]:
```suggestion
        "title": qsTr("Zoom"),
```

File: src/qml/qgismobileapp.qml [hunk 14]
```diff
@@ -3931,4 +3932,44 @@ ApplicationWindow {
...
+        "description": qsTr("In addition to the pinch gesture, these buttons help you quickly zoom in and out."),
+        "target": () => [zoomToolbar]
+      }, {
+        "title": qsTr("GNSS"),
```
Comment (nirvn) [inline src/qml/qgismobileapp.qml:3950]:
```suggestion
        "title": qsTr("Positioning"),
```

File: src/qml/qgismobileapp.qml (hunk 12 above)
Comment (mohsenD98) [reply src/qml/qgismobileapp.qml:3942]: :+1:

File: src/qml/qgismobileapp.qml (hunk 13 above)
Comment (mohsenD98) [reply src/qml/qgismobileapp.qml:3946]: :+1:

File: src/qml/qgismobileapp.qml (hunk 14 above)
Comment (mohsenD98) [reply src/qml/qgismobileapp.qml:3950]: :+1:

**Review by nirvn — APPROVED** [review]

**Pull request merged**